### PR TITLE
Add Wallet Connect base implementation of the connection flow

### LINKED ISSUE #19907: Add the base implementation for WC connection
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a developer, I want to have a way to set up Wallet Connect sessions, so that we have a shared base implementation of the flow

### Description

In order to not diverge/conflict when implementing the features, we should have a starting point from which we will add the functionality on top from other user stories.

### Acceptance Criteria

- [x] should run the Wallet Connect initialisation (behind a feature flag) on login
- [x] should be able to request a connection to the test dApp (https://react-app.walletconnect.com/) with at least one chain (via REPL)
- [x] should be able to approve the connection request (via REPL)
- [x] should have step-by-step instructions for testing the connection flow

### Notes

[comment]: # (For the developer who will implement. Include planning poker estimates, guidance, etc.)

### PR DESCRIPTION
**Pull request by clauxx**
fixes #19907

### Summary

In order to not diverge/conflict when implementing the features, we should have a starting point from which we will add the functionality on top from other user stories. Currently connection only works with Optimism mainnet.

### Steps to test (developers)

1. Make sure the wallet connect is enabled via feature flags (set `FLAG_WALLET_CONNECT_ENABLED` to `1` in the `.env.debug` file or hardcode the default value)
2. Add `WALLET_CONNECT_PROJECT_ID` to .env.debug (ask for the ID in the #wallet-connect Discord channel) or hardcode the value
3. Go to https://react-app.walletconnect.com/
4. Un-toggle "Testnets Only?"
5. Select only Optimism and press connect at the bottom
6. Copy the uri and run in the REPL: `(rf/dispatch [:wallet-connect/pair uri])`
7. When successful, run: `(rf/dispatch [:wallet-connect/approve-session])`
8. The connection should be established

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/wallet_connect/effects.cljs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+(ns status-im.contexts.wallet.wallet-connect.effects
+  (:require [re-frame.core :as rf]
+            [status-im.contexts.wallet.wallet-connect.utils :as wallet-connect.utils]
+            [taoensso.timbre :as log]))
+
+(rf/reg-fx
+ :effects.wallet-connect/init
+ (fn [{:keys [on-success on-fail]}]
+   (-> (wallet-connect.utils/init)
+       (.then on-success)
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/wallet_connect/effects.cljs]: This is clean but as a team we decided to shift to promesa.

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+(rf/reg-event-fx
+ :wallet-connect/approve-session
+ (fn [{:keys [db]}]
+   ;; NOTE: hardcoding optimism for the base implementation
+   (let [crosschain-ids       [constants/optimism-crosschain-id]
+         web3-wallet          (get db :wallet-connect/web3-wallet)
+         current-proposal     (get db :wallet-connect/current-proposal)
+         accounts             (get-in db [:wallet :accounts])
+         address              (-> accounts keys first)
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs:70]: This should also have a ⚠️ note ⚠️, or it should chose the selected account, not the first account.

File: src/status_im/contexts/wallet/wallet_connect/events.cljs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ :wallet-connect/register-event-listeners
+ (fn [{:keys [db]}]
+   (let [web3-wallet (get db :wallet-connect/web3-wallet)]
+     {:fx [[:effects.wallet-connect/register-event-listener
+            web3-wallet
+            constants/wallet-connect-session-proposal-event
+            #(rf/dispatch [:wallet-connect/on-session-proposal %])]]})))
+
+(rf/reg-event-fx
+ :wallet-connect/on-init-fail
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/wallet_connect/events.cljs:31]: We can clear the proposal from db if initialisation fails. Stale state is bad for health.

**Review by shivekkhurana — CHANGES_REQUESTED** [review]

File: src/status_im/contexts/wallet/wallet_connect/events.cljs (hunk 2 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/wallet_connect/events.cljs:70]: done

File: src/status_im/contexts/wallet/wallet_connect/events.cljs (hunk 3 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/wallet_connect/events.cljs:31]: agreed, but resetting the proposal should happen when "sessionApproved" fails/succeeds, so added it there.

File: src/status_im/contexts/wallet/wallet_connect/effects.cljs (hunk 1 above)
Comment (clauxx) [reply src/status_im/contexts/wallet/wallet_connect/effects.cljs]: done

**Review by shivekkhurana — APPROVED** [review]
image
LG

**Review by alwx — APPROVED** [review]
🚀🚀🚀

**Comment by mariia-skrypnyk** [comment]
Hi @clauxx !

E2E are known and not related.

**Pull request merged**